lib/actions/parse_csv.ex

```diff
@@ -0,0 +1,85 @@
+defmodule WorkflowEngine.Actions.ParseCsv do
+  @moduledoc """
+  Executes a request against the internal API using BXDK.
+  """
+  use OK.Pipe
+
+  @behaviour WorkflowEngine.Action
+
+  @impl WorkflowEngine.Action
+
+  NimbleCSV.define(CSVComma,
+    separator: ",",
+    escape: "\""
+  )
+
+  NimbleCSV.define(CSVSemiColon,
+    separator: ";",
+    escape: "\""
+  )
+
+  NimbleCSV.define(CSVTab,
+    separator: "\t",
+    escape: "\""
+  )
+
+  def execute(state, %{"type" => "parse_csv", "csv_settings" => csv_settings} = step) do
+    with {:ok, data} <- get_data(step, state) do
+      csv_module = get_csv_module(csv_settings)
+
+      rows =
+        data
+        |> trim_bom()
+        |> List.wrap()
+        |> csv_module.to_line_stream()
+        |> csv_module.parse_stream(skip_headers: false)
+
+      header_row = Stream.take(rows, 1) |> Enum.at(0)
+
+      content_rows =
+        Stream.drop(rows, 1)
+        |> Stream.map(fn row ->
+          Enum.zip(header_row, row)
+          |> Map.new()
+        end)
+
+      {:ok, {state, content_rows}}
+    else
+      {:error, reason} ->
+        {:error, reason}
```

Comment:
In this case, the `else` branch isn't needed - when it doesn't exist, the `with` will pass the error as-is.
Handling and returning the error makes sense if there's additional code to be done in between (like logging and error)